piksi_tools/udp_bridge.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Fergus Noble <fergus@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+the :mod:`sbp.client.examples.mavproxybridge` module contains an example of reading SBP
+messages from a serial port, filtering for observations, and sending them over udp
+to a mavproxy instance for transmission to an ArduCopter quad.
+"""
+
+from sbp.client.drivers.pyserial_driver import PySerialDriver
+from sbp.client.handler import Handler
+from sbp.observation import SBP_MSG_OBS, SBP_MSG_BASE_POS
+
+import socket
+import struct
+import time
+
+OBS_MSGS = [SBP_MSG_OBS,
+            SBP_MSG_BASE_POS]
+
+DEFAULT_SERIAL_PORT = "/dev/ttyUSB0"
+DEFAULT_SERIAL_BAUD = 1000000
+
+DEFAULT_UDP_ADDRESS = "127.0.0.1"
+DEFAULT_UDP_PORT    = 13320
+
+def get_args():
+  """
+  Get and parse arguments.
+  """
+  import argparse
+  parser = argparse.ArgumentParser(description="Swift Navigation SBP Example.")
+  parser.add_argument("-s", "--serial-port",
+                      default=[DEFAULT_SERIAL_PORT], nargs=1,
+                      help="specify the serial port to use.")
+  parser.add_argument("-b", "--baud",
+                      default=[DEFAULT_SERIAL_BAUD], nargs=1,
+                      help="specify the baud rate to use.")
+  parser.add_argument("-a", "--address",
+                      default=[DEFAULT_UDP_ADDRESS], nargs=1,
+                      help="specify the UDP IP Address to use.")
+  parser.add_argument("-p", "--udp-port",
+                      default=[DEFAULT_UDP_PORT], nargs=1,
+                      help="specify the UDP Port to use.")
+  return parser.parse_args()
+
+def open_socket():
+  return socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
```

Comment:
The one thing that is hard from a CLI perspective is exactly how to filter messages.  What if you only want to broadcast 10 SBP Messages over UDP?  What if you want to filter the rate of some of the messages broadcast? I think it would be pretty hard to make a full featured command line tool and we might be better off just providing examples for people to do there own python.